pkg/networkdevice/metadata/payload_utils_test.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package metadata
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/common"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_batchPayloads(t *testing.T) {
+	collectTime := common.MockTimeNow()
+	deviceID := "123"
+	devices := []DeviceMetadata{
+		{ID: deviceID},
+	}
+
+	var interfaces []InterfaceMetadata
+	for i := 0; i < 350; i++ {
+		interfaces = append(interfaces, InterfaceMetadata{DeviceID: deviceID, Index: int32(i)})
+	}
+	var ipAddresses []IPAddressMetadata
+	for i := 0; i < 100; i++ {
+		ipAddresses = append(ipAddresses, IPAddressMetadata{InterfaceID: deviceID + ":1", IPAddress: "1.2.3.4", Prefixlen: 24})
+	}
+	var topologyLinks []TopologyLinkMetadata
+	for i := 0; i < 100; i++ {
+		topologyLinks = append(topologyLinks, TopologyLinkMetadata{
+			Local:  &TopologyLinkSide{Interface: &TopologyLinkInterface{ID: "a"}},
+			Remote: &TopologyLinkSide{Interface: &TopologyLinkInterface{ID: "b"}},
+		})
+	}
+	payloads := BatchPayloads("my-ns", "127.0.0.0/30", collectTime, 100, devices, interfaces, ipAddresses, topologyLinks)
+
+	assert.Equal(t, 6, len(payloads))
```

Comment:
You might want to use `require.Len(t, payloads, 6)` instead to to avoid panic if we have less than 6 items.